site/static/docs/4.3/assets/scss/_colors.scss

```diff
@@ -15,3 +15,103 @@
+.bd-orange-800 { color: color-yiq($orange-800); background-color: $orange-800; }
+.bd-orange-900 { color: color-yiq($orange-900); background-color: $orange-900; }
+
+.bd-yellow-100 { color: color-yiq($yellow-100); background-color: $yellow-100; }
+.bd-yellow-200 { color: color-yiq($yellow-200); background-color: $yellow-200; }
+.bd-yellow-300 { color: color-yiq($yellow-300); background-color: $yellow-300; }
+.bd-yellow-400 { color: color-yiq($yellow-400); background-color: $yellow-400; }
+.bd-yellow-500 { color: color-yiq($yellow-500); background-color: $yellow-500; }
+.bd-yellow-600 { color: color-yiq($yellow-600); background-color: $yellow-600; }
+.bd-yellow-700 { color: color-yiq($yellow-700); background-color: $yellow-700; }
+.bd-yellow-800 { color: color-yiq($yellow-800); background-color: $yellow-800; }
+.bd-yellow-900 { color: color-yiq($yellow-900); background-color: $yellow-900; }
+
+.bd-green-100 { color: color-yiq($green-100); background-color: $green-100; }
+.bd-green-200 { color: color-yiq($green-200); background-color: $green-200; }
+.bd-green-300 { color: color-yiq($green-300); background-color: $green-300; }
+.bd-green-400 { color: color-yiq($green-400); background-color: $green-400; }
+.bd-green-500 { color: color-yiq($green-500); background-color: $green-500; }
+.bd-green-600 { color: color-yiq($green-600); background-color: $green-600; }
+.bd-green-700 { color: color-yiq($green-700); background-color: $green-700; }
+.bd-green-800 { color: color-yiq($green-800); background-color: $green-800; }
+.bd-green-900 { color: color-yiq($green-900); background-color: $green-900; }
+
+.bd-teal-100 { color: color-yiq($teal-100); background-color: $teal-100; }
+.bd-teal-200 { color: color-yiq($teal-200); background-color: $teal-200; }
+.bd-teal-300 { color: color-yiq($teal-300); background-color: $teal-300; }
+.bd-teal-400 { color: color-yiq($teal-400); background-color: $teal-400; }
+.bd-teal-500 { color: color-yiq($teal-500); background-color: $teal-500; }
+.bd-teal-600 { color: color-yiq($teal-600); background-color: $teal-600; }
+.bd-teal-700 { color: color-yiq($teal-700); background-color: $teal-700; }
+.bd-teal-800 { color: color-yiq($teal-800); background-color: $teal-800; }
+.bd-teal-900 { color: color-yiq($teal-900); background-color: $teal-900; }
+
+.bd-cyan-100 { color: color-yiq($cyan-100); background-color: $cyan-100; }
+.bd-cyan-200 { color: color-yiq($cyan-200); background-color: $cyan-200; }
+.bd-cyan-300 { color: color-yiq($cyan-300); background-color: $cyan-300; }
+.bd-cyan-400 { color: color-yiq($cyan-400); background-color: $cyan-400; }
+.bd-cyan-500 { color: color-yiq($cyan-500); background-color: $cyan-500; }
+.bd-cyan-600 { color: color-yiq($cyan-600); background-color: $cyan-600; }
+.bd-cyan-700 { color: color-yiq($cyan-700); background-color: $cyan-700; }
+.bd-cyan-800 { color: color-yiq($cyan-800); background-color: $cyan-800; }
+.bd-cyan-900 { color: color-yiq($cyan-900); background-color: $cyan-900; }
```

Comment:
Could use something like this (colors listed explicit as ``$colors`` contains grays)
```scss
@each $color in ("blue", "indigo", "purple", "pink", "red", "orange", "yellow", "green", "teal", "cyan") {
  @for $i from 1 through 9 {
    .bd-#{$color}-#{$i * 100} {
      color: color-yiq(#{#{$color}-100});
      background-color: #{#{$color}-100};
    }
  }
}
```

Reply:
I don't think this 'll work. But if it does, could you provide a demo?

Reply:
You're right ... I dunno how, but while writing this I must have imagined it working at some point :blush: Sorry for that!

Only posibility I'd see is to use a map instead of plain variables. That drops overridability of the single shades and tints. On the other hand I'm asking myself how often people really would want to change a single color variant when using these shades & tints, so they could just override the whole map.

Actually working example:
```scss
@function shades-and-tints($colors: $colors) {
  $shadesAndTints: ();

  @each $name, $color in $colors {
    $shadesAndTints: map-merge(
      (
        "#{$name}": (
          "100": shade-color($color, 8),
          "200": shade-color($color, 6),
          "300": shade-color($color, 4),
          "400": shade-color($color, 2),
          "500": $color,
          "600": tint-color($color, 2),
          "700": tint-color($color, 4),
          "800": tint-color($color, 6),
          "900": tint-color($color, 8)
        )
      ),
      $shadesAndTints
    );
  }

  @return $shadesAndTints;
}

$colorVariants: shades-and-tints() !default;

@each $colorName, $variants in $colorVariants {
  @each $variant, $colorValue in $variants {
    .bd-#{$colorName}-#{$variant} {
      background-color: #{$colorValue};
      color: color-yiq($colorValue);
    }
  }
}
```